src/Temporalio/Bridge/EnvConfig.cs

```diff
@@ -0,0 +1,353 @@
+                    };
+
+                    var result = Interop.Methods.temporal_core_client_config_load(&options);
+                    return ProcessAllProfilesResult(runtime, result);
+                }
+            }
+            catch (JsonException ex)
+            {
+                throw new InvalidOperationException($"Failed to deserialize client config: {ex.Message}", ex);
+            }
+            finally
+            {
+                foreach (var ptr in strings)
+                {
+                    Marshal.FreeHGlobal(ptr);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Loads a specific client configuration profile.
+        /// </summary>
+        /// <param name="runtime">Runtime to use.</param>
+        /// <param name="profile">Profile name to load.</param>
+        /// <param name="source">Data source to load configuration from.</param>
+        /// <param name="disableFile">If true, do not load from file (only from environment).</param>
+        /// <param name="disableEnv">If true, disable environment variable overrides.</param>
+        /// <param name="configFileStrict">If true, fail if configuration file is invalid.</param>
+        /// <param name="overrideEnvVars">Environment variables to use, or null to use system environment.</param>
+        /// <returns>Client configuration for the specified profile.</returns>
+        public static ClientEnvConfig.ConfigProfile LoadClientConfigProfile(
+            Runtime runtime,
+            string? profile,
+            DataSource? source,
+            bool disableFile = false,
+            bool disableEnv = false,
+            bool configFileStrict = false,
+            IReadOnlyDictionary<string, string>? overrideEnvVars = null)
+        {
+            using var scope = new Scope();
+            var strings = new List<IntPtr>();
+
+            try
+            {
+                var profilePtr = profile != null ? Marshal.StringToHGlobalAnsi(profile) : IntPtr.Zero;
+                if (profilePtr != IntPtr.Zero)
+                {
+                    strings.Add(profilePtr);
+                }
+
+                var pathPtr = IntPtr.Zero;
+                if (source != null && !string.IsNullOrEmpty(source.Path))
+                {
+                    pathPtr = Marshal.StringToHGlobalAnsi(source.Path);
+                    strings.Add(pathPtr);
+                }
+
+                var envVarsRef = overrideEnvVars?.Count > 0
+                    ? scope.ByteArray(JsonSerializer.Serialize(overrideEnvVars, JsonOptions))
+                    : ByteArrayRef.Empty.Ref;
+
+                unsafe
+                {
+                    var options = new Interop.TemporalCoreClientConfigProfileLoadOptions
+                    {
+                        profile = (sbyte*)profilePtr,
+                        path = (sbyte*)pathPtr,
+                        data = source?.Data != null ? scope.ByteArray(source.Data) : ByteArrayRef.Empty.Ref,
```

Comment:
```suggestion
                        data = scope.ByteArray(source?.Data),
```

It's already built to handle nulls